lib/matplotlib/tests/test_legend.py

```diff
@@ -294,6 +294,21 @@ def test_legend_remove():
     assert ax.get_legend() is None
 
 
+def test_reverse_legend():
+    fig, ax = plt.subplots()
+    x = [1, 2, 3]
+    y = [1, 2, 3], [2, 3, 4], [3, 4, 5]
+    labels = ["First label", "Second label", "Third label"]
+
+    ax.plot(x, y, label=labels)
+    leg = ax.legend(reverse=True)
+    actual_labels = []
+    for text in leg.get_texts():
+        actual_labels.append(text.get_text())
+    excepted_labels = list(reversed(labels))
+    assert actual_labels == excepted_labels
```

Comment:
You can shorten this to:

```suggestion
    actual_labels = [t.get_gext() for t in leg.get_texts()]
    assert actual_labels == list(reversed(labels))
```

Additionally you should check the markers, which you can get via
```
markers = [h.get_marker() for h in leg.legend_handles]
```
I think you have to unroll the `plot` command into three separate commands to be able to specify markers explicitly (we intentionally don't support vectorization of keyword arguments in `plot`).

Please also note that `leg.legend_handles` was just added today to the main branch, so you need to rebase on `main` to be able to use that. - Let us know if you need help with rebasing.